packages/request-logic/src/request-logic.ts

```diff
@@ -640,14 +641,18 @@ export default class RequestLogic implements RequestLogicTypes.IRequestLogic {
             pending: requestAndMeta.pending,
             request: requestAndMeta.request,
           });
-
           // workaround to quiet the error "finalResult.meta.ignoredTransactions can be undefined" (but defined in the initialization value of the accumulator)
           (finalResult.meta.ignoredTransactions || []).push(requestAndMeta.ignoredTransactions);
 
           // add the transactionManagerMeta
           (finalResult.meta.transactionManagerMeta || []).push(
             requestAndMeta.transactionManagerMeta,
           );
+
+          finalResult.meta.pagination = {
+            ...finalResult.meta.pagination,
+            ...requestAndMeta.pagination,
+          };
```

Comment:
Can you explain what this is doing?  I think you're keeping a running total of the pagination data, overwriting the old pagination with the new pagination.

Reply:
You can see above that the reduce is merging all the requests and meta in one object.